independent-projects/junit5-virtual-threads/src/main/java/io/quarkus/test/junit5/virtual/internal/Collector.java

```diff
@@ -22,146 +22,132 @@ public class Collector implements Consumer<RecordedEvent> {
-                }
-                var end = System.nanoTime();
-                LOGGER.log(Level.FINE, "Event collection stopped in {0}s", (end - begin) / 1000000);
-                return new ArrayList<>(events);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RuntimeException(e);
+        try {
+            if (!latch.await(10, TimeUnit.SECONDS)) {
+                throw new IllegalStateException("Unable to start JFR collection, STOP_EVENT event not received after 10s");
             }
+            var end = System.nanoTime();
+            LOGGER.log(Level.FINE, "Event collection stopped in {0}s", (end - begin) / 1000000);
+            return new ArrayList<>(events);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
         }
-        return Collections.emptyList();
     }
 
     public void shutdown() {
-        if (facade != null) {
-            CountDownLatch latch = new CountDownLatch(1);
-            var begin = System.nanoTime();
-            observers.add(re -> {
-                if (re.getEventType().getName().equals(InternalEvents.SHUTDOWN_EVENT_NAME)) {
-                    latch.countDown();
-                    return true;
-                }
-                return false;
-            });
-            InternalEvents.ShutdownEvent event = new InternalEvents.ShutdownEvent();
-            event.commit();
-            try {
-                if (latch.await(10, TimeUnit.SECONDS)) {
-                    state = State.INIT;
-                    var end = System.nanoTime();
-                    LOGGER.log(Level.FINE, "Event collector shutdown in {0}s", (end - begin) / 1000000);
-                    facade.stop();
-                } else {
-                    throw new IllegalStateException(
-                            "Unable to stop JFR collection, RecordingStoppedEvent event not received at 10s");
-                }
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RuntimeException(e);
+        CountDownLatch latch = new CountDownLatch(1);
+        var begin = System.nanoTime();
+        observers.add(re -> {
+            if (re.getEventType().getName().equals(InternalEvents.SHUTDOWN_EVENT_NAME)) {
+                latch.countDown();
+                return true;
+            }
+            return false;
+        });
+        InternalEvents.ShutdownEvent event = new InternalEvents.ShutdownEvent();
+        event.commit();
+        try {
+            if (latch.await(10, TimeUnit.SECONDS)) {
+                state = State.INIT;
+                var end = System.nanoTime();
+                LOGGER.log(Level.FINE, "Event collector shutdown in {0}s", (end - begin) / 1000000);
+                recordingStream.close();
```

Comment:
facade tried to call `stop` method (`tempStop = lookup.findVirtual(clazz, "stop", MethodType.methodType(boolean.class));`), but that's not present in RecordingStream class

https://github.com/openjdk/jdk/blob/master/src/jdk.jfr/share/classes/jdk/jfr/consumer/RecordingStream.java#L351
says `* To stop the stream, use the {@link #close()} method.`